tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -0,0 +1,551 @@
+
+    def fill_branch_name_template(self, template: str, ref: str,
+                                  kind: str) -> str:
+        """Build a branch name based on a template.
+
+        Replace '{}' by ref if present (otherwise return template unchanged).
+        ref must be a branch, tag or sha.
+        kind is a short, human-readable description of the template's purpose.
+        """
+        if '{}' not in template:
+            return template
+        if not self.ref_looks_ok_for_template(ref):
+            self.log_error('Cannot create branch name for {} from {}',
+                           kind, ref)
+            raise ValueError('Bad ref {} for {} branch name template {}'
+                             .format(ref, kind, template))
+        return template.replace('{}', ref)
+
+    def list_commit_range(self, start: str, last: str) -> List[str]:
+        """List the commits from start to last, excluding last."""
+        commits = self.get_git_lines(['rev-list', start + '..' + last])
+        return list(reversed(commits))
+
+    def system_is_ok(self) -> bool:
+        """Check that the required tools are present and the working directory is suitable."""
+        ok = True
+        # 'git worktree remove' was added in Git 2.17.0. It's the most
+        # recent Git feature that we need.
+        output = self.run_git(['worktree', '--help'])
+        if 'git worktree remove' not in output:
+            self.log_error('Your version of Git is too old. This script needs Git >=2.17.0.')
+            ok = False
+        # Check that we're in a Git working directory. Remember its path.
+        output = self.run_git(['rev-parse', '--show-toplevel'])
+        self.original_worktree_path = output.rstrip('\n')
+        self.info('Git root: {}', self.original_worktree_path)
+        return ok
+
+    def generate_worktree_path(self, arg: str) -> str:
+        """Generate the path to a temporary worktree.
+
+        This function only constructs a path, it does not create the worktree.
+        """
+        assert self.original_worktree_path is not None
+        worktree_dir = 'tmp-{arg}-{pid}'.format(arg=arg, pid=os.getpid())
+        return os.path.join(os.path.dirname(self.original_worktree_path),
+                            worktree_dir)
+
+    def cleanup_worktree(self) -> None:
+        """Remove the temporary worktree.
+
+        Do nothing if the temporary worktree does not exist.
+        """
+        worktree_path = self.worktree_path
+        self.worktree_path = None
+        if worktree_path is None: # not named yet
+            return
+        if not os.path.exists(worktree_path): # not created yet
+            self.info('No worktree to clean up: {}', worktree_path)
+            return
+        self.info('Removing worktree: {}', worktree_path)
+        self.run_git(['worktree', 'remove', worktree_path])
+
+    def create_backup_branch(self, template: str, ref: str) -> None:
+        """Create a branch new_name pointing to ref.
```

Comment:
Minor: `new_name` is not an argument to this method, so it feels a bit weird to see it in the docstring.